clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -1629,37 +1630,79 @@ void FunctionVariableConstraint::brainTransplant(ConstraintVariable *FromCV,
   auto RetVar = getOnly(returnVars);
   RetVar->brainTransplant(FromRetVar, I);
   // Transplant params.
-  assert(From->numParams() == numParams());
-  for (unsigned i = 0; i < From->numParams(); i++) {
-    CVarSet &FromP = From->getParamVar(i);
-    CVarSet &P = getParamVar(i);
-    auto FromVar = getOnly(FromP);
-    auto Var = getOnly(P);
-    Var->brainTransplant(FromVar, I);
+  if (numParams() == From->numParams()) {
+    for (unsigned i = 0; i < From->numParams(); i++) {
+      CVarSet &FromP = From->getParamVar(i);
+      CVarSet &P = getParamVar(i);
+      auto FromVar = getOnly(FromP);
+      auto Var = getOnly(P);
+      Var->brainTransplant(FromVar, I);
+    }
+  } else if (numParams() != 0 && From->numParams() == 0) {
+    auto &CS = I.getConstraints();
+    vector<ParamDeferment> &defers = From->getDeferredParams();
+    assert(getDeferredParams().size() == 0);
+    for(auto deferred : defers ) {
+      assert(numParams() == deferred.PS.size());
+      for(unsigned i = 0; i < deferred.PS.size(); i++) {
+        CVarSet ParamDC = getParamVar(i);
+        CVarSet ArgDC = deferred.PS[i];
+        constrainConsVarGeq(ParamDC, ArgDC, CS, &(deferred.PL), Wild_to_Safe, false, &I);
+      }
+    }
+  } else {
+    llvm_unreachable("Brain transplanting to function w/ empty param vars");
   }
 }
 
-void FunctionVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV) {
+void FunctionVariableConstraint::mergeDeclaration(ConstraintVariable *FromCV,
+                                                  ProgramInfo &I) {
+  // `this`: is the declaration the tool saw first
+  // `FromCV`: is the declaration seen second, it cannot have defered constraints
   FVConstraint *From = dyn_cast<FVConstraint>(FromCV);
   assert (From != nullptr);
+  assert(From->getDeferredParams().size() == 0);
   // Transplant returns.
   auto FromRetVar = getOnly(From->getReturnVars());
   auto RetVar = getOnly(returnVars);
-  RetVar->mergeDeclaration(FromRetVar);
+  RetVar->mergeDeclaration(FromRetVar, I);
+  bool isEmpty = numParams() == 0;
```

Comment:
I think you can delete `isEmpty` -- you refer to `numParams() == 0` directly, below.